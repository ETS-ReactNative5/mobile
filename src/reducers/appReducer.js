import * as ActionConstants from '../constants/actions';


const InitialAppState = {
    device: {
        width: 0,
        height: 0
    }
};

export default function app(state=InitialAppState, action) {
    switch (action.type) {
        case ActionConstants.UPDATE_SCREEN_DIMENSIONS: {
            return {...state, device: action.dimensions}
        }
        default: {
            return state
        }
    }
}