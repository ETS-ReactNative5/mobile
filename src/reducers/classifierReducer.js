import * as ActionConstants from '../constants/actions';
import R from 'ramda'

const InitialClassifier = {
    currentWorkflowId: 0,
    workflow: {},
    tasks: {},
    tutorial: {},
    project: {},
    needsTutorial: {},
    classification: {},
    guide: {},
    isFetching: false,
    isSuccess: false,
    isFailure: false,
    annotations: {},
    upcomingSubjects: {},
    seenThisSession: {},
    subject: {},
    nextSubject: {},
    questionContainerHeight: {},
    inPreviewMode: false
};

export default function classifier(state=InitialClassifier, action) {
    switch (action.type) {
        case ActionConstants.ADD_CLASSIFIER_TUTORIAL: {
            const workflowIdLens = R.lensProp(action.workflowId)
            return { ...state, tutorial: R.set(workflowIdLens, action.tutorial, state.tutorial) }
        }
        case ActionConstants.ADD_WORKFLOW_NEEDS_TUTORIAL: {
            const workflowIdLens = R.lensProp(action.workflowId)
            return { ...state, needsTutorial: R.set(workflowIdLens, action.needsTutorial, state.needsTutorial) }
        }
        case ActionConstants.SET_CLASSIFIER_GUIDE: {
            const workflowIdLens = R.lensProp(action.workflowId)
            return { ...state, guide: R.set(workflowIdLens, action.guide, state.guide) }
        }
        case ActionConstants.SET_CLASSIFICATION: {
            const workflowIdLens = R.lensProp(action.workflowId)
            return { ...state, classification: R.set(workflowIdLens, action.classification, state.classification) }
        }
        case ActionConstants.INITIALIZE_ANNOTATION: {
            const workflowIdLens = R.lensProp(action.workflowId)
            return { ...state, annotations: R.set(workflowIdLens, {}, state.annotations)} 
        }
        case ActionConstants.ADD_ANNOTATION_TO_TASK: {
            const taskLens = R.lensPath([action.workflowId, action.task])
            if (action.asList) {
                const annotationList = R.view(taskLens, state.annotations) || []
                annotationList.push(action.annotation)
                return { ...state, annotations: R.set(taskLens, annotationList, state.annotations) }
            }
            return { ...state, annotations: R.set(taskLens, action.annotation, state.annotations) } 
        }
        case ActionConstants.REMOVE_ANNOTATION_FROM_TASK: {
            const taskLens = R.lensPath([action.workflowId, action.task])
            const annotationList = R.view(taskLens, state.annotations) || []
            const removedAnnotationList = R.reject(R.equals(action.annotation), annotationList)
            return { ...state, annotations: R.set(taskLens, removedAnnotationList, state.annotations) }
        }
        case ActionConstants.SET_UPCOMING_SUBJECTS: {
            const workflowIdLens = R.lensProp(action.workflowId)
            return { ...state, upcomingSubjects: R.set(workflowIdLens, action.upcomingSubjects, state.upcomingSubjects)} 
        }
        case ActionConstants.SET_SUBJECT_SEEN_THIS_SESSION: {
            const workflowIdLens = R.lensProp(action.workflowId)
            const subjectsSeenArray = state.seenThisSession[action.workflowId] || []
            subjectsSeenArray.push(action.subjectId)
            return { ...state, seenThisSession: R.set(workflowIdLens, subjectsSeenArray, state.seenThisSession)} 
        }
        case ActionConstants.SET_SUBJECT: {
            const workflowIdLens = R.lensProp(action.workflowId)
            return { ...state, subject: R.set(workflowIdLens, action.subject, state.subject)}
        }
        case ActionConstants.SET_NEXT_SUBJECT: {
            const workflowIdLens = R.lensProp(action.workflowId)
            return { ...state, nextSubject: R.set(workflowIdLens, action.nextSubject, state.nextSubject)}
        }
        case ActionConstants.SET_QUESTION_CONTAINER_HEIGHT: {
            const workflowIdLens = R.lensProp(action.workflowId)
            return { ...state, questionContainerHeight: R.set(workflowIdLens, action.questionContainerHeight, state.questionContainerHeight)}
        }
        case ActionConstants.REQUEST_CLASSIFIER_DATA: {
            return { ...state, isFetching: true }
        }
        case ActionConstants.CLASSIFIER_DATA_SUCCESS: {
            return { ...state, isFetching: false, isSuccess: true }
        }
        case ActionConstants.CLASSIFIER_DATA_FAILURE: {
            return { ...state, isFetching: false, isFailure: true }
        }
        case ActionConstants.CLEAR_CLASSIFIER_DATA: {
            return { ...state, isFetching: false, isSuccess: false, isFailure: false }
        }
        case ActionConstants.SET_CLASSIFIER_TEST_MODE: {
            return { ...state, inPreviewMode: action.isTestMode}
        }
        default:
            return state;
    }
}