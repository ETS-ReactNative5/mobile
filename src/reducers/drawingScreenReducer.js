import * as ActionConstants from '../constants/actions';
import R from 'ramda'

const InitialState = {
    shapes: {},
    idCount: 0
}

export default function drawingScreen(state=InitialState, action) {
    switch (action.type) {
        case (ActionConstants.ADD_SHAPE): {
            const newShapeObject = R.set(R.lensProp(state.idCount), action.shapeObject, state.shapes)
            // const newShapesArray = R.append(action.shapeObject, state.shapes)
            return { ...state, shapes: newShapeObject, idCount: state.idCount + 1 }
        }
        case (ActionConstants.REMOVE_SHAPE): {
            const newShapesArray = R.omit([action.index], state.shapes)
            return { ...state, shapes: newShapesArray }
        }
        case (ActionConstants.MUTATE_SHAPE): {
            const { dx, dy, dw, dh, index } = action
            const updatedShapes = R.set(R.lensProp(index),updateShape(dx, dy, dw, dh)(state.shapes[index]), state.shapes)
            return { ...state, shapes: updatedShapes}
        }
        default:
            return state
    }
}

const updateShape = ( dx, dy, dw, dh) => {
    return (shape) => ({
        ... shape,
        x: shape.x + dx,
        y: shape.y + dy,
        width: shape.width + dw,
        height: shape.height + dh
    })
}