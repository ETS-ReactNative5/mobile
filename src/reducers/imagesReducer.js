import * as ActionConstants from '../constants/actions'
import R from 'ramda'

const InitialState = {}

export default function images(state=InitialState, action) {
    switch (action.type) {
        case ActionConstants.SAVE_IMAGE_LOCATION: {
            const imageKeyLens = R.lensProp(action.remoteSource)
            return R.set(imageKeyLens, action.localSource, state)
        }
        case ActionConstants.DELETE_IMAGE_LOCATION: {
            return R.reject((value) => value === action.localSource, state)
        }
        case ActionConstants.CLEAR_IMAGE_CACHE: {
            return InitialState
        }
        default:
            return state
    }
}