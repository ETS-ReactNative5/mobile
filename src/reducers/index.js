import { merge, lensPath, set } from 'ramda'

export const InitialState = {
  user: {},
  registration: { global_email_communication: true },
  isFetching: false,
  errorMessage: null,
  isConnected: null,
  selectedDiscipline: null,
  projectList: {},
  projectWorkflows: {},
  webViewNavCounter: 0,
  notificationProject: {},
  notificationPayload: {},
  notifications: { general: true },
  session: {},
  settings: { promptForWorkflow: false },
  pushEnabled: false,
  pushPrompted: false,
  classifier: { //key is workflow ID for each
    isFetching: true,
    currentWorkflowID: 0,
    workflow: {},
    project: {},
    tutorial: {},
    needsTutorial: {},
    questionContainerHeight: {},
  },
}

export default function(state=InitialState, action) {
  switch (action.type) {
    case 'SET_STATE':
      return set(lensPath(action.stateKey.split('.')), action.value, state)
    case 'SET_USER':
      return merge(state, {
        user: action.user
      })
    case 'SET_IS_FETCHING':
      return merge(state, {
        isFetching: action.isFetching
      })
    case 'SET_ERROR':
      return merge(state, {
        errorMessage: action.errorMessage
      })
    case 'SET_PROJECT_LIST':
      return merge(state, {
        projectList: action.projectList
      })
    default:
      return InitialState;
  }
}
