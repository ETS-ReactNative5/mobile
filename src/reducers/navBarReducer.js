import * as ActionConstants from '../constants/actions';
import * as R from 'ramda'

const InitialNavBarState = {
    titles: {},
    backgroundColors: {}
};

export default function navBar(state=InitialNavBarState, action) {
    switch (action.type) {
        case ActionConstants.SET_NAVBAR_TITLE: {
            const titles = R.set(R.lensProp(action.pageKey), action.title, state.titles);
            return { ...state, titles }
        }
        case ActionConstants.SET_NAVBAR_COLOR: {
            const backgroundColors = R.set(R.lensProp(action.pageKey), action.color, state.backgroundColors);
            return { ...state, backgroundColors}
        }
        case ActionConstants.SET_NAVBAR_COLOR_TO_DEFAULT: {
            const backgroundColors = state.backgroundColors
            delete backgroundColors[action.pageKey]
            return { ...state, backgroundColors}
        }
        default:
            return state;
    }
}