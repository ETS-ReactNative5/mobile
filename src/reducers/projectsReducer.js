import * as ActionConstants from '../constants/actions';
import * as R from 'ramda';

const InitialProjectState = {
    isLoading: false,
    isSuccess: false,
    isFailure: false,
    projectList: [],
    previewProjectList: [],
    collaboratorIds: [],
    ownerIds: []
};

export default function projects(state=InitialProjectState, action) {
    switch (action.type) {
        case ActionConstants.PROJECTS_REQUEST: {
            return {
                ...state,
                isLoading: true
            }
        }
        case ActionConstants.PROJECTS_SUCCESS: {
            return {
                ...state,
                isLoading: false,
                isSuccess: true,
                isFailure: false,
            };
        }
        case ActionConstants.ADD_PROJECTS: {
            const { arePreview, projects } = action
            return arePreview ? { ...state, previewProjectList: projects } : { ...state, projectList: projects}
        }
        case ActionConstants.PROJECTS_FAILURE: {
            return {
                ...state,
                isLoading: false,
                isFailure: true
            }
        }
        case ActionConstants.ADD_OWNER_PROJECT_ID: {
            return {
                ...state,
                ownerIds: R.append(action.projectId, state.ownerIds)
            }
        }
        case ActionConstants.ADD_COLLABORATOR_PROJECT_ID: {
            return {
                ...state,
                collaboratorIds: R.append(action.projectId, state.collaboratorIds)
            }
        }
        case ActionConstants.SIGN_OUT: {
            return InitialProjectState;
        }
        default:
            return state;
    }
}
