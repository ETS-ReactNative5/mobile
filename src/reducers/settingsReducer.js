import * as ActionConstants from '../constants/actions'
import R from 'ramda'

const InitialState = {
    notificationsInitialized: false,
    showAllWorkflows: true,
    enableNotifications: true,
    newProjectNotifications: true,
    newBetaNotifications: true,
    urgentHelpNotification: true,
    projectSpecificNotifications: []
}

const markProjectsSubscriptionStatus = (projectId, subscribe) => {
    return (project) => {
        if (project.id === projectId) {
            project.subscribed = subscribe
        }
        return project
    }
}

const hasProjectId = (projectId) => {
    return (subbedProject) => subbedProject.id === projectId
}

export default function settings(state=InitialState, action) {
    switch (action.type) {
        case ActionConstants.UPDATE_SUBSCRIPTION_TO_PROJECT: {
            const { subscribe, projectId } = action
            const newProjectSubscriptions = state.projectSpecificNotifications.map(markProjectsSubscriptionStatus(projectId, subscribe))
            return { ...state, projectSpecificNotifications: newProjectSubscriptions }
        }
        case ActionConstants.ADD_PROJECT_SUBSCRIPTION: {
            action.project.subscribed = true
            return { ...state, projectSpecificNotifications: R.append(action.project, state.projectSpecificNotifications)}
        }
        case ActionConstants.REMOVE_PROJECT_SUBSCRIPTION: {
            const projectSpecificNotifications = R.reject(hasProjectId(action.project.id), state.projectSpecificNotifications)
            return { ...state, projectSpecificNotifications }
        }
        case ActionConstants.UPDATE_SHOW_ALL_WORKLFOWS:
            return { ...state, showAllWorkflows: action.enabled }
        case ActionConstants.UPDATE_ENABLE_NOTIFICATIONS:
            return { ...state, enableNotifications: action.enabled }
        case ActionConstants.UPDATE_NEW_PROJECT_NOTIICATIONS:
            return { ...state, newProjectNotifications: action.enabled }
        case ActionConstants.UPDATE_BETA_NOTIFICATIONS:
            return { ...state, newBetaNotifications: action.enabled}
        case ActionConstants.UPDATE_URGENT_HELP_NOTIFICATIONS:
            return { ...state, urgentHelpNotification: action.enabled}
        case ActionConstants.NOTIFICATIONS_INITIALIZED: 
            return { ...state, notificationsInitialized: true}
        default:
            return state
    }
}