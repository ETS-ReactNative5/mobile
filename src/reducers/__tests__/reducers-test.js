import 'react-native'
import reducers, { initialState } from '../index.js'
import { setUser, setState } from '../../actions/index'

it('sets user', () => {
  let fakeUser = { avatar: 'Some avatar', display_name: 'Me'}
  expect(reducers(initialState, setUser(fakeUser))).toMatchSnapshot()
})

it('unsets user', () => {
  expect(reducers(initialState, setUser({}))).toMatchSnapshot()
})

it('sets state from initial state', () => {
  expect(reducers(initialState, setState('isConnected', true))).toMatchSnapshot()
})

it('sets a new state', () => {
  expect(reducers(initialState, setState('bobLoblaw', 'Has a law blog'))).toMatchSnapshot()
})
