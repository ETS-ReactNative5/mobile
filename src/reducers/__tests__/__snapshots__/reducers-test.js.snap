// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets a new state 1`] = `
Object {
  "classifier": Object {
    "annotations": Object {},
    "classification": Object {},
    "currentWorkflowId": 0,
    "guide": Object {},
    "inPreviewMode": false,
    "isFailure": false,
    "isFetching": false,
    "isSuccess": false,
    "needsTutorial": Object {},
    "nextSubject": Object {},
    "project": Object {},
    "questionContainerHeight": Object {},
    "seenThisSession": Object {},
    "subject": Object {},
    "tasks": Object {},
    "tutorial": Object {},
    "upcomingSubjects": Object {},
    "workflow": Object {},
  },
  "main": Object {
    "bobLoblaw": "Has a law blog",
    "errorMessage": null,
    "isConnected": null,
    "isFetching": false,
    "notificationPayload": Object {},
    "notificationProject": Object {},
    "notifications": Object {
      "general": true,
    },
    "pushEnabled": false,
    "pushPrompted": false,
    "registration": Object {
      "global_email_communication": true,
    },
    "selectedDiscipline": null,
    "session": Object {},
    "settings": Object {
      "promptForWorkflow": false,
    },
    "webViewNavCounter": 0,
  },
  "navBar": Object {
    "backgroundColors": Object {},
    "titles": Object {},
  },
  "projects": Object {
    "collaboratorIds": Array [],
    "isFailure": false,
    "isLoading": false,
    "isSuccess": false,
    "ownerIds": Array [],
    "previewProjectList": Array [],
    "projectList": Array [],
  },
  "user": Object {},
}
`;

exports[`sets state from initial state 1`] = `
Object {
  "classifier": Object {
    "annotations": Object {},
    "classification": Object {},
    "currentWorkflowId": 0,
    "guide": Object {},
    "inPreviewMode": false,
    "isFailure": false,
    "isFetching": false,
    "isSuccess": false,
    "needsTutorial": Object {},
    "nextSubject": Object {},
    "project": Object {},
    "questionContainerHeight": Object {},
    "seenThisSession": Object {},
    "subject": Object {},
    "tasks": Object {},
    "tutorial": Object {},
    "upcomingSubjects": Object {},
    "workflow": Object {},
  },
  "main": Object {
    "errorMessage": null,
    "isConnected": true,
    "isFetching": false,
    "notificationPayload": Object {},
    "notificationProject": Object {},
    "notifications": Object {
      "general": true,
    },
    "pushEnabled": false,
    "pushPrompted": false,
    "registration": Object {
      "global_email_communication": true,
    },
    "selectedDiscipline": null,
    "session": Object {},
    "settings": Object {
      "promptForWorkflow": false,
    },
    "webViewNavCounter": 0,
  },
  "navBar": Object {
    "backgroundColors": Object {},
    "titles": Object {},
  },
  "projects": Object {
    "collaboratorIds": Array [],
    "isFailure": false,
    "isLoading": false,
    "isSuccess": false,
    "ownerIds": Array [],
    "previewProjectList": Array [],
    "projectList": Array [],
  },
  "user": Object {},
}
`;

exports[`sets user 1`] = `
Object {
  "classifier": Object {
    "annotations": Object {},
    "classification": Object {},
    "currentWorkflowId": 0,
    "guide": Object {},
    "inPreviewMode": false,
    "isFailure": false,
    "isFetching": false,
    "isSuccess": false,
    "needsTutorial": Object {},
    "nextSubject": Object {},
    "project": Object {},
    "questionContainerHeight": Object {},
    "seenThisSession": Object {},
    "subject": Object {},
    "tasks": Object {},
    "tutorial": Object {},
    "upcomingSubjects": Object {},
    "workflow": Object {},
  },
  "main": Object {
    "errorMessage": null,
    "isConnected": null,
    "isFetching": false,
    "notificationPayload": Object {},
    "notificationProject": Object {},
    "notifications": Object {
      "general": true,
    },
    "pushEnabled": false,
    "pushPrompted": false,
    "registration": Object {
      "global_email_communication": true,
    },
    "selectedDiscipline": null,
    "session": Object {},
    "settings": Object {
      "promptForWorkflow": false,
    },
    "user": Object {
      "avatar": "Some avatar",
      "display_name": "Me",
    },
    "webViewNavCounter": 0,
  },
  "navBar": Object {
    "backgroundColors": Object {},
    "titles": Object {},
  },
  "projects": Object {
    "collaboratorIds": Array [],
    "isFailure": false,
    "isLoading": false,
    "isSuccess": false,
    "ownerIds": Array [],
    "previewProjectList": Array [],
    "projectList": Array [],
  },
  "user": Object {},
}
`;

exports[`unsets user 1`] = `
Object {
  "classifier": Object {
    "annotations": Object {},
    "classification": Object {},
    "currentWorkflowId": 0,
    "guide": Object {},
    "inPreviewMode": false,
    "isFailure": false,
    "isFetching": false,
    "isSuccess": false,
    "needsTutorial": Object {},
    "nextSubject": Object {},
    "project": Object {},
    "questionContainerHeight": Object {},
    "seenThisSession": Object {},
    "subject": Object {},
    "tasks": Object {},
    "tutorial": Object {},
    "upcomingSubjects": Object {},
    "workflow": Object {},
  },
  "main": Object {
    "errorMessage": null,
    "isConnected": null,
    "isFetching": false,
    "notificationPayload": Object {},
    "notificationProject": Object {},
    "notifications": Object {
      "general": true,
    },
    "pushEnabled": false,
    "pushPrompted": false,
    "registration": Object {
      "global_email_communication": true,
    },
    "selectedDiscipline": null,
    "session": Object {},
    "settings": Object {
      "promptForWorkflow": false,
    },
    "user": Object {},
    "webViewNavCounter": 0,
  },
  "navBar": Object {
    "backgroundColors": Object {},
    "titles": Object {},
  },
  "projects": Object {
    "collaboratorIds": Array [],
    "isFailure": false,
    "isLoading": false,
    "isSuccess": false,
    "ownerIds": Array [],
    "previewProjectList": Array [],
    "projectList": Array [],
  },
  "user": Object {},
}
`;
